src/main/scala/it/pagopa/interop/authorizationmanagement/api/impl/ClientApiServiceImpl.scala

```diff
@@ -155,162 +118,88 @@ final case class ClientApiServiceImpl(
     consumerId: Option[String],
     purposeId: Option[String],
     kind: Option[PersistentClientKind]
-  ): LazyList[PersistentClient] = {
+  ): Try[LazyList[PersistentClient]] = {
     @tailrec
     def readSlice(
       commander: EntityRef[Command],
       from: Int,
       to: Int,
       lazyList: LazyList[PersistentClient]
-    ): LazyList[PersistentClient] = {
-      lazy val slice: Seq[PersistentClient] =
-        Await
-          .result(
-            commander.ask(ref => ListClients(from, to, relationshipId, consumerId, purposeId, kind, ref)),
-            Duration.Inf
-          )
-          .getValue
-      if (slice.isEmpty)
-        lazyList
-      else
-        readSlice(commander, to, to + sliceSize, slice.to(LazyList) #::: lazyList)
+    ): Try[LazyList[PersistentClient]] = {
+      lazy val slice: Try[Seq[PersistentClient]] =
```

Comment:
not investing too much time here since probably all listings will be moved to read model, but at least the catch of possible exceptions has been added to gracefully handle them